docker/docker.bzl

```diff
@@ -30,55 +30,65 @@ load(":push.bzl", "docker_push")
 # The release of the github.com/google/containerregistry to consume.
 CONTAINERREGISTRY_RELEASE = "v0.0.11"
 
-def docker_repositories():
+def docker_repositories(excludes=[], overrides={}):
   """Download dependencies of docker rules."""
-  native.http_file(
-    name = "puller",
-    url = ("https://storage.googleapis.com/containerregistry-releases/" +
-           CONTAINERREGISTRY_RELEASE + "/puller.par"),
-    sha256 = "90a76d01ee57a5df7353a533c96966822c9efda61636c11eac11344171556017",
-    executable = True,
-  )
+  gcr_release = overrides.get("containerregistry", CONTAINERREGISTRY_RELEASE)
+  if "puller" not in excludes:
+    native.http_file(
+      name = "puller",
+      url = ("https://storage.googleapis.com/containerregistry-releases/" +
+             gcr_release + "/puller.par"),
+      sha256 = overrides.get(
```

Comment:
You could make it optional but generally this go against reproducibility which is going to get back at you.

BTW, maybe use native.existing_rule to make sure you do not add an existing deps instead, that would better address #55.